sklearn/_engine/base.py

```diff
@@ -112,3 +112,29 @@ def get_engine_classes(engine_name, default, verbose=False):
                 f"trying engine {engine_class.__module__}.{engine_class.__qualname__} ."
             )
         yield provider, engine_class
+
+
+def convert_attributes(method):
+    """Convert estimator attributes after calling the decorated method
+
+    The attributes of an estimator can be stored in "engine native" types
+    (default) or "scikit-learn native" types. Decorate methods, like `fit`
+    that set attributes.
+    """
+
+    @wraps(method)
+    def wrapper(self, *args, **kwargs):
+        r = method(self, *args, **kwargs)
+        convert_attributes = get_config()["engine_attributes"]
+
+        if convert_attributes == "sklearn_types":
+            engine = self._engine_class
+            for name, attribute in vars(self).items():
+                # Using __dlpack__ as signal for "not a basic Python type"
+                if hasattr(attribute, "__dlpack__"):
```

Comment:
```suggestion
                # Using __dlpack__ as signal for "not a basic Python type"
                if hasattr(attribute, "__dlpack__"):
```

Maybe this choice (to convert or passthrough) should be left to the engine itself because it can know the specifics of the estimator better than the generic decorator (in particular by using the name).

Reply:
I like that suggestion. I ended up using `__dlpack__` because I couldn't find anything good. But I didn't think of just passing everything to the engine and leaving it up to it to decide.